Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-22-254-49.ec2.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 618a1231acc28da8293f3af3eadb35b3cdc810c5
Containerized: False
VCPKG build disabled.
Number of Error(s): 5
/home/centos/smoketest/PR-16216/HPCC-Platform/system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:430:1: error: control reaches end of non-void function [-Werror=return-type]
cc1plus: some warnings being treated as errors
make[2]: *** [system/logaccess/Azure/LogAnalytics/CurlClient/CMakeFiles/AzureLogAnalyticsCurllogaccess.dir/build.make:76: system/logaccess/Azure/LogAnalytics/CurlClient/CMakeFiles/AzureLogAnalyticsCurllogaccess.dir/AzureLogAnalyticsCurlClient.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:7546: system/logaccess/Azure/LogAnalytics/CurlClient/CMakeFiles/AzureLogAnalyticsCurllogaccess.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-06-16_19-05-49
time logs started
Prep time   :    62 sec (00:01:02)
Build time  :   132 sec (00:02:12)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   194 sec (00:03:14)
time logs end

[Build error:](https://gist.github.com/63b71284fbbe64ac2a9b9449104ff706)
Build log [Build log](https://gist.github.com/HPCCSmoketest/63b71284fbbe64ac2a9b9449104ff706#file-result-22-06-16-19-02-19-log)

--- a/system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp
+++ b/system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp
@@ -423,6 +423,12 @@ void AzureLogAnalyticsCurlClient::azureLogAnalyticsTimestampQueryRangeString(Str
     }
 }
 
+void throwIfMultiIndexDetected(const char * currentIndex, const char * proposedIndex)
+{
+    if (!isEmptyString(currentIndex) && !strsame(currentIndex,proposedIndex))
+        throw makeStringExceptionV(-1, "%s: Multi-index query not supported: '%s' - '%s'", COMPONENT_NAME, currentIndex, proposedIndex);
+}
+
 void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryString, StringBuffer & queryIndex, const ILogAccessFilter * filter)
 {
     if (filter == nullptr)
@@ -443,8 +449,7 @@ void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryStr
 
         if (!m_workunitIndexSearchPattern.isEmpty())
         {
-            if (!queryIndex.isEmpty() && strsame(queryIndex, m_workunitIndexSearchPattern.str()))
-                throw makeStringExceptionV(-1, "%s: Multi-index query not supported: '%s' - '%s'", COMPONENT_NAME, queryIndex.str(), m_workunitIndexSearchPattern.str());
+            throwIfMultiIndexDetected(queryIndex.str(), m_workunitIndexSearchPattern.str());
             queryIndex = m_workunitIndexSearchPattern;
         }
 
@@ -460,8 +465,7 @@ void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryStr
 
         if (!m_classIndexSearchPattern.isEmpty())
         {
-            if (!queryIndex.isEmpty() && strsame(queryIndex.str(), m_classIndexSearchPattern.str()))
-                throw makeStringExceptionV(-1, "%s: Multi-index query not supported: '%s' - '%s'", COMPONENT_NAME, queryIndex.str(), m_classIndexSearchPattern.str());
+            throwIfMultiIndexDetected(queryIndex.str(), m_classIndexSearchPattern.str());
             queryIndex = m_classIndexSearchPattern.str();
         }
 
@@ -477,10 +481,8 @@ void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryStr
 
         if (!m_audienceIndexSearchPattern.isEmpty())
         {
-            if (!queryIndex.isEmpty() && strsame(queryIndex.str(), m_audienceIndexSearchPattern.str()))
-                throw makeStringExceptionV(-1, "%s: Multi-index query not supported: '%s' - '%s'", COMPONENT_NAME, queryIndex.str(), m_audienceIndexSearchPattern.str());
-
-            queryIndex = m_audienceIndexSearchPattern.str();
+           throwIfMultiIndexDetected(queryIndex.str(), m_audienceIndexSearchPattern.str());
+           queryIndex = m_audienceIndexSearchPattern.str();
         }
 
         DBGLOG("%s: Searching log entries by target audience: '%s'...", COMPONENT_NAME, queryValue.str());
@@ -495,10 +497,8 @@ void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryStr
 
         if (!m_componentsIndexSearchPattern.isEmpty())
         {
-            if (!queryIndex.isEmpty() && strsame(queryIndex.str(), m_componentsIndexSearchPattern.str()))
-                throw makeStringExceptionV(-1, "%s: Multi-index query not supported: '%s' - '%s'", COMPONENT_NAME, queryIndex.str(), m_componentsIndexSearchPattern.str());
-
-            queryIndex = m_componentsIndexSearchPattern.str();
+           throwIfMultiIndexDetected(queryIndex.str(), m_componentsIndexSearchPattern.str());
+           queryIndex = m_componentsIndexSearchPattern.str();
         }
 
         DBGLOG("%s: Searching '%s' component log entries...", COMPONENT_NAME, queryValue.str());
@@ -513,9 +513,7 @@ void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryStr
 
         if (!m_hostIndexSearchPattern.isEmpty())
         {
-            if (!queryIndex.isEmpty() && strsame(queryIndex.str(), m_hostIndexSearchPattern.str()))
-                throw makeStringExceptionV(-1, "%s: Multi-index query not supported: '%s' - '%s'", COMPONENT_NAME, queryIndex.str(), m_hostIndexSearchPattern.str());
-
+            throwIfMultiIndexDetected(queryIndex.str(), m_hostIndexSearchPattern.str());
             queryIndex = m_hostIndexSearchPattern.str();
         }
 
@@ -531,9 +529,7 @@ void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryStr
 
         if (!m_instanceIndexSearchPattern.isEmpty())
         {
-            if (!queryIndex.isEmpty() && strsame(queryIndex.str(), m_instanceIndexSearchPattern.str()))
-                throw makeStringExceptionV(-1, "%s: Multi-index query not supported: '%s' - '%s'", COMPONENT_NAME, queryIndex.str(), m_instanceIndexSearchPattern.str());
-
+            throwIfMultiIndexDetected(queryIndex.str(),  m_instanceIndexSearchPattern.str());
             queryIndex = m_instanceIndexSearchPattern.str();
         }
 

